test/unit/vectortiles.js

```diff
@@ -66,159 +67,239 @@ describe('Vector tiles', function () {
-        it('throws an error because no style was provided', () => {
-            assert.throws(() => new VectorTilesSource({}), {
-                name: 'Error',
-                message: 'New VectorTilesSource: style is required',
-            });
+describe('VectorTilesSource', function () {
+    it('throws an error because no style was provided', () => {
+        assert.throws(() => new VectorTilesSource({}), {
+            name: 'Error',
+            message: 'New VectorTilesSource: style is required',
         });
+    });
 
-        it('reads tiles URL from the style', (done) => {
-            const source = new VectorTilesSource({
-                style: {
-                    sources: { geojson: { tiles: ['http://server.geo/{z}/{x}/{y}.pbf'] } },
-                    layers: [],
-                },
-            });
-            source.whenReady.then(() => {
-                // eslint-disable-next-line no-template-curly-in-string
-                assert.equal(source.url, 'http://server.geo/${z}/${x}/${y}.pbf');
-                done();
-            }).catch(done);
+    it('reads tiles URL directly from the style', (done) => {
+        const source = new VectorTilesSource({
+            style: {
+                sources: { sourceTiles: { tiles: ['http://server.geo/{z}/{x}/{y}.pbf'] } },
+                layers: [],
+            },
         });
+        source.whenReady.then(() => {
+            assert.equal(source.urls.size, 1);
+            // eslint-disable-next-line no-template-curly-in-string
+            assert.ok(source.urls.has('http://server.geo/${z}/${x}/${y}.pbf'));
+            done();
+        })
+            .catch(done);
+    });
 
-        it('reads the background layer', (done) => {
-            const source = new VectorTilesSource({
-                url: 'fakeurl',
-                style: {
-                    sources: { geojson: {} },
-                    layers: [{ type: 'background' }],
-                },
-            });
-            source.whenReady.then(() => {
-                assert.ok(source.backgroundLayer);
-                done();
-            }).catch(done);
+    // need fetcher !
+    // it('reads tiles URL from an url', (done) => {
+    //     const source = new VectorTilesSource({
+    //         style: {
+    //             sources: { sourceUrl: { url: 'mapbox://mapbox.mapbox-terrain-v2,mapbox.mapbox-streets-v7' } },
+    //             layers: [],
+    //         },
+    //         accessToken: 'pk.eyJ1IjoiZnRvcm9tYW5vZmYiLCJhIjoiY2xrc2Zpa2o3MDQxNTNxcG5sczJyaTlncyJ9.5KFKdMLIiy-b_fAjSVhjCQ',
+    //     });
+    //     source.whenReady
+    //         .then(() => {
+    //             assert.equal(source.url.size, 2);
+    //             // eslint-disable-next-line no-template-curly-in-string
+    //             assert.ok(source.url.has('http://server.geo/${z}/${x}/${y}.pbf'));
+    //             done();
+    //         })
+    //         .catch(done);
+    // });
+
```

Comment:
Is it a todo ? Maybe link an issue to it

Reply:
I did change the test for it to work, but it will need the Need PR #2183 to be working (to be able to mock the fetcher).
I reworked it but let it in comment for the moment, to avoid the import of the sinon library in this PR.